torch/_inductor/choices.py

```diff
@@ -40,6 +47,26 @@ class MyHeuristics(InductorChoices):
             torch._inductor.virtualized.V.set_choices_handler(MyHeuristics())
     """
 
+    def __init__(self):
+        self.config_heuristics = self._get_device_config_heuristic()
+
+    def _get_device_config_heuristic(self, device_type="cuda"):
+        from torch._inductor.utils import get_gpu_type
+
+        device_type = get_gpu_type()
+
+        if device_type == "cuda":
+            if torch.version.hip is None:
+                return CUDAConfigHeuristic()
+            else:
+                return ROCmConfigHeuristic()
+        elif device_type == "xpu":
+            return XPUConfigHeuristic()
+        elif torch.cuda.is_available():
+            return BaseConfigHeuristic()
+        else:
+            return CPUConfigHeuristic()
```

Comment:
This won't work for programs that use multiple devices (most commonly GPU+CPU).  We should be selecting the hursitics based on the device of the op, not the hardware properties of the machine.  Maybe:
```
def get_device_config_heuristic(self, device):
   ... similar to the code you have ...
```